charts/opentelemetry-collector/templates/configmap-agent.yaml

```
@@ -1,4 +1,4 @@
-{{- if .Values.agentCollector.enabled -}}
+{{- if and (not .Values.standaloneCollector.enabled) (not (eq .Values.mode "deployment")) (or .Values.agentCollector.enabled (eq .Values.mode "daemonset")) -}}
```

Why we need too many conditions here? This shouldn't be a place to validate deployment mode. 

```suggestion
{{- if or .Values.agentCollector.enabled (eq .Values.mode "daemonset") -}}
```

---

Since `.Values.agentCollector.enabled` is true by default and we want `mode: deployment` to override it I believe we need to check that `mode != deployment`, but I believe I can remove the check against `.Values.standaloneCollector.enabled`.  That was there because mode was being set to daemonset by default.